http/client/request.go

```diff
@@ -81,10 +81,21 @@ func NewFileRequest(L *lua.LState) int {
 		return
 	}
 
-	if err := writeFile(files, writer); err != nil{
-		files.ForEach(func(k, v lua.LValue) {
-			writeFile(v.(*lua.LTable), writer)
-		})
+	var err error
+	if files.Len() == 0 {
+		err = writeFile(files, writer)
+	} else {
+		for key, value := files.Next(lua.LNil); key != lua.LNil; key, value = files.Next(key) {
+			err = writeFile(value.(*lua.LTable), writer)
+			if err != nil {
+				break
+			}
+		}
+	}
+	if err != nil {
+		L.Push(lua.LNil)
```

Comment:
Yes, this looks good! I like this return style - clients can then check and report or handle the err, if any

Reply:
I do see this pattern repeated however… it's fine, but if you see way that could easily refactor this to call a more go-styled function that just returned err when != nil , then you could just check that result and push the nil, err once

Reply:
There is currently no better way to refactor these repeated errors.😅